File: valid_palindrone/gpt_sol.js
function isPalindrome(s) {
    // Convert the string to lowercase and remove non-alphanumeric characters
    let cleanString = s.toLowerCase().replace(/[^a-z0-9]/g, "");

    // Check if the cleaned string is a palindrome
    let left = 0;
    let right = cleanString.length - 1;
    while (left < right) {
        if (cleanString[left] !== cleanString[right]) {
            return false;
        }
        left++;
        right--;
    }

    // If the loop completes without returning false, it's a palindrome
    return true;
}

// Test the function
console.log(isPalindrome("A man, a plan, a canal: Panama")); // true
console.log(isPalindrome("race a car")); // false
console.log(isPalindrome(" ")); // true

File: practice-algos/valid_palindrone/gpt_sol.py
def is_palindrome(s):
    # Convert the string to lowercase and remove non-alphanumeric characters
    clean_string = "".join(char.lower() for char in s if char.isalnum())

    # Check if the cleaned string is a palindrome
    return clean_string == clean_string[::-1]

# Test the function
print(is_palindrome("A man, a plan, a canal: Panama")) # True
print(is_palindrome("race a car")) # False
print(is_palindrome(" ")) # True